src/Microsoft.ML.Data/Model/Repository.cs

```diff
@@ -109,13 +109,35 @@ internal Repository(bool needDir, IExceptionContext ectx)
             _open = new List<Entry>();
             if (needDir)
             {
-                DirTemp = GetTempPath();
+                DirTemp = GetShortTempPath();
                 Directory.CreateDirectory(DirTemp);
             }
             else
                 GC.SuppressFinalize(this);
         }
 
+        private static string GetShortTempPath()
+        {
+            Guid guid = Guid.NewGuid();
+            var guidName = guid.ToString();
+            int index = guidName.IndexOf('-');
+            if (index < 0)
+                index = Math.Min(8, guidName.Length);
+            guidName = guidName.Substring(0, index);
+            var path = Path.Combine(Path.GetTempPath(), "TLC_" + guidName);
+            while (Directory.Exists(path))
```

Comment:
>Exists [](start = 29, length = 6)

why even bother with GUId?, why not just take a random int and convert to hex? That's the shortest and will collide only once in a billion times on average (assuming you randomize properly.)